File: Java/oop/OOP/src/belajar/java/data/Person.java
package belajar.java.data;

public class Person {
    public String name;
    public String address;
    public final String country = "United States";

    public Person(String name, String address) {
        this.name = name;
        this.address = address;
    }

    public Person(String paramName) {
        this(paramName, null);
    }

    public Person() {
        this(null, null);
    }

    public void sayHello(String name) {
        System.out.println("Hello " + name + " My name is " + this.name);
    }
}

File: programming-language/PHP/programerzamannow/pembelajaran/oop/OOP/data/Person.php
<?php

class Person
{
    public const AUTHOR = "Gendhi Ramona";
    public string $name;
    public ?string $address = null;
    public string $country = "Indonesia";

    public function __construct(string $name, ?string $address)
    {
        $this->name = $name;
        $this->address = $address;
    }

    public function sayHello(?string $name)
    {
        if (is_null($name)) {
            echo "Hi, my name is $this->name" . PHP_EOL;
        } else {
            echo "Hi $name, my name is $this->name" . PHP_EOL;
        }
    }

    public function info()
    {
        echo "Author : " . self::AUTHOR . PHP_EOL;
    }

    public function __destruct()
    {
        echo "Object Person {$this->name} has been destroyed" . PHP_EOL;
    }
}
